Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIBookmarks.mm

```diff
@@ -0,0 +1,154 @@
+    @"description": @"bookmarkpermission off Test",
+    @"version": @"1",
+
+    @"permissions": @[],
+    @"background": @{
+        @"scripts": @[ @"background.js", ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+};
+
+static auto *bookmarkOnManifest = @{
+    @"manifest_version": @3,
+    @"name": @"bookmarkpermission on Test",
+    @"description": @"bookmarkpermission on Test",
+    @"version": @"1",
+
+    @"permissions": @[ @"bookmarks" ],
+    @"background": @{
+        @"scripts": @[ @"background.js", ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+    @"action": @{
+        @"default_title": @"Test Action",
+        @"default_popup": @"popup.html",
+    },
+};
+
+
+#pragma mark - Test Fixture
+
+class WKWebExtensionAPIBookmarks : public testing::Test {
+protected:
+    WKWebExtensionAPIBookmarks()
+    {
+        NSLog(@"[DEBUG] WKWebExtensionAPIBookmarks constructor: START");
+        bookmarkConfig = WKWebExtensionControllerConfiguration.defaultConfiguration;
+        if (!bookmarkConfig.webViewConfiguration)
+            bookmarkConfig.webViewConfiguration = [[WKWebViewConfiguration alloc] init];
+
+        BOOL featureEnabledSuccessfully = NO;
+        _WKFeature* targetFeature = nil;
+
+        for (_WKFeature *feature in [WKPreferences _features]) {
+            if ([feature.key isEqualToString:@"WebExtensionBookmarksEnabled"]) {
+                targetFeature = feature;
+                // Attempt to enable
+                [bookmarkConfig.webViewConfiguration.preferences _setEnabled:YES forFeature:feature];
+                // Verify immediately if it was set
+                if ([bookmarkConfig.webViewConfiguration.preferences _isEnabledForFeature:feature])
+                    featureEnabledSuccessfully = YES;
+
+                break; // Found the feature, no need to continue loop
+            }
+        }
+
+        if (targetFeature) {
+            NSLog(@"[DEBUG] WebExtensionBookmarksEnabled feature flag was targeted. Attempted to enable. Current state: %s",
+            featureEnabledSuccessfully ? "YES" : "NO (or enabling failed/not reflected immediately)");
+        } else
+            NSLog(@"[DEBUG] WebExtensionBookmarksEnabled feature flag NOT FOUND in [WKPreferences _features].");
+
+
+    }
+    RetainPtr<TestWebExtensionManager> getManagerFor(NSArray<NSString *> *script, NSDictionary<NSString *, id> *manifest)
+    {
+        return getManagerFor(@{ @"background.js" : Util::constructScript(script) }, manifest);
+    }
+
+    RetainPtr<TestWebExtensionManager> getManagerFor(NSDictionary<NSString *, id> *resources, NSDictionary<NSString *, id> *manifest)
+    {
+        return Util::parseExtension(manifest, resources, bookmarkConfig);
+    }
+
+    WKWebExtensionControllerConfiguration *bookmarkConfig;
+    RetainPtr<WKWebExtension> extension;
+};
+
+#pragma mark - Common Tests
+
+TEST_F(WKWebExtensionAPIBookmarks, APISUnavailableWhenManifestDoesNotRequest)
```

Comment:
i think this test name is a bit misleading. the API shouldn't be available just because the manifest requests it. granting the permission on line 146 is what makes it available